src/hooks/useBanner.ts

```diff
@@ -0,0 +1,181 @@
+import { useMemo } from 'react';
+import dayjs from 'dayjs';
+import timezone from 'dayjs/plugin/timezone';
+import utc from 'dayjs/plugin/utc';
+import yaml from 'js-yaml';
+import * as yup from 'yup';
+import { BANNER_CONTENT_FILE, BANNER_NAMEPSACE } from '~/consts/banner';
+import { useK8sWatchResource } from '~/k8s';
+import { ConfigMap } from '~/types/configmap';
+import { bannerConfigYupSchema } from '~/utils/validation-utils';
+import { ConfigMapGroupVersionKind, ConfigMapModel } from './../models/config-map';
+
+dayjs.extend(utc);
+dayjs.extend(timezone);
+
+export type BannerType = 'info' | 'warning' | 'danger';
+export type BannerConfig = {
+  summary: string;
+  type: BannerType;
+  year?: string;
+  month?: string;
+  startTime?: string;
+  endTime?: string;
+  timeZone?: string;
+  dayOfWeek?: number; // 0-6 for Sunday-Saturday
+  dayOfMonth?: number; // 1-31 for the day of the month
+};
+
+export function inferRepeatType(banner: BannerConfig): 'weekly' | 'monthly' | 'none' {
```

Comment:
Create enum for `weekly`, 'monthly`, `none`.

Reply:
updated